test/e2e_node/pod_conditions_test.go

```diff
@@ -134,6 +134,55 @@ func runPodFailingConditionsTest(f *framework.Framework, hasInitContainers, chec
 		// Verify PodReady is not set (since sandboxcreation is blocked)
 		_, err = getTransitionTimeForPodConditionWithStatus(p, v1.PodReady, false)
 		framework.ExpectNoError(err)
+
+		ginkgo.By("checking pod condition for a pod when volumes is created")
+
+		p2 := webserverPodSpec("pod-"+string(uuid.NewUUID()), "web1", "init1", hasInitContainers)
+		p2.Spec.Volumes = []v1.Volume{
+			{
+				Name: "cm",
+				VolumeSource: v1.VolumeSource{
+					ConfigMap: &v1.ConfigMapVolumeSource{
+						LocalObjectReference: v1.LocalObjectReference{Name: "cm-does-not-exist"},
+					},
+				},
+			},
+		}
+		p2.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
+			{
+				Name:      "cm",
+				MountPath: "/config",
+			},
+		}
+
+		p2 = e2epod.NewPodClient(f).Create(ctx, p2)
```

Comment:
I think you want to create the config map before you create this pod.  

1) create configmap
2) create pod that references config-map
3) verify true condition.

If that doesn't work, I'd suggest trying to test this flow manually and see if there is something wrong.

Reply:
move the configmap create before pod. waiting for CI.